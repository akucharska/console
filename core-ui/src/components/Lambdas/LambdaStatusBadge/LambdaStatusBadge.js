import React from 'react';

import { StatusBadge } from 'react-shared';

import {
  LAMBDA_PHASES,
  LAMBDA_ERROR_PHASES,
} from 'components/Lambdas/constants';

import { statusType } from 'components/Lambdas/helpers/lambdas';
import { formatMessage } from 'components/Lambdas/helpers/misc';

export function LambdaStatusBadge({ status }) {
  const latestStatus = status.conditions[0]; // TODO Translate status like in console backend components/console-backend-service/internal/domain/serverless/function_converter.go
  const translatedStatus =
    latestStatus.status === 'True' && latestStatus.type === 'Running'
      ? { phase: 'RUNNING', reason: null, message: null }
      : { phase: 'FAILED', reason: null, message: null };

  const statusPhase = translatedStatus.phase;

  const texts = LAMBDA_PHASES[statusPhase];
  let badgeType = statusType(statusPhase);
  if (badgeType === 'info') {
    badgeType = undefined;
  }

  let tooltipText;

  if (LAMBDA_ERROR_PHASES.includes(statusPhase)) {
    const formattedError = formatMessage(LAMBDA_PHASES.ERROR_SUFFIX, {
      error: translatedStatus.message,
    });
    tooltipText = `${texts.MESSAGE} ${formattedError}`;
  }

  return (
    <StatusBadge tooltipContent={tooltipText} type={badgeType}>
      {texts.TITLE}
    </StatusBadge>
  );
}
