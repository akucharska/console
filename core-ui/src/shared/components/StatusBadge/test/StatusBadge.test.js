import renderer from 'react-test-renderer';
import StatusBadge from '../StatusBadge';
import LambdaStatusBadge from '../LambdaStatusBadge';
import React from 'react';

describe('StatusBadge', () => {
  it('should render loading state initially', () => {
    const component = renderer.create(<StatusBadge status="INITIAL" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('LambdaStatusBadge', () => {
  it('LambdaStatusBadge should render with status building', () => {
    const component = renderer.create(<LambdaStatusBadge status="Building" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
