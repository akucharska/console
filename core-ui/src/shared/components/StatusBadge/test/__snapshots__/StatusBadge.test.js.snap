// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LambdaStatusBadge LambdaStatusBadge should render with status building 1`] = `
<span
  className="fd-badge"
>
  Building
</span>
`;

exports[`StatusBadge should render loading state initially 1`] = `
<span
  className="fd-badge"
>
  INITIAL
</span>
`;
